src/box/box.cc

```diff
@@ -2832,10 +2835,19 @@ box_wait_quorum(uint32_t lead_id, int64_t target_lsn, int quorum,
 		trigger_create(&t.base, box_quorum_on_ack_f, NULL, NULL);
 		trigger_add(&replicaset.on_ack, &t.base);
 		double deadline = ev_monotonic_now(loop()) + timeout;
-		do {
-			if (fiber_yield_deadline(deadline))
+		while (!fiber_is_cancelled() && t.ack_count < t.quorum) {
+			if (fiber_cond_wait_deadline(&synchro_quorum_cond,
+						     deadline) == -1) {
 				break;
-		} while (!fiber_is_cancelled() && t.ack_count < t.quorum);
+			}
+
+			if (quorum != replication_synchro_quorum) {
+				diag_set(ClientError, ER_QUORUM_WAIT, quorum,
+					 "replication_synchro_quorum was changed while waiting");
```

Comment:
Let's make the msg consistent with:

https://github.com/tarantool/tarantool/blob/a0748bea849bd9e069b1be817b9323fd87240838/src/box/box.cc#L3221-L3222

Reply:
Nah, I meant `quorum was changed`, but now I understood, that this may be not obvious, which quorum was changed (maybe it's `replication_connect_quorum`), so it's there, where the message is incorrect. Here we can leave the msg here as it's now